src/main/java/com/endava/webapp/validation/UniqueConstraintValidation.java

```diff
@@ -0,0 +1,19 @@
+package com.endava.webapp.validation;
+
+import com.endava.webapp.exception.UniqueConstraintViolationException;
+import com.endava.webapp.repository.EmployeeRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
+public class UniqueConstraintValidation {
+
+    private final EmployeeRepository employeeRepository;
+
+    public void employeeEmailOrPhoneAlreadyExistsValidation(String email, String phoneNumber) {
+        if (employeeRepository.existsByEmailOrPhoneNumber(email, phoneNumber)) {
+            throw new UniqueConstraintViolationException("Employee's email and phone number must be unique");
+        }
+    }
+}
```

Comment:
I hope that you extracted this logic in a separate component because you've read **S**OLID principles, otherwise it's very interesting where did you get such ideas from :)

Anyway if you wanna to achieve this logic:
1. Probably it's not that bad to do this right in the Service implementation
2. Looks like hibernate has inbuilt functionality for this one - https://www.baeldung.com/jpa-unique-constraints

P.S. regarding 2nd point - make sure not to break anything else by this change ))

Reply:
Discussed via Teams. Not applicable right now, probably because of h2 database usage, but well noted for future encounters.